packages/recoil/recoil_values/__tests__/Recoil_atom-test.js

```diff
@@ -911,6 +911,39 @@ describe('Effects', () => {
     expect(c.textContent).toEqual('"DEFAULT_A""DEFAULT_B"');
   });
 
+  testRecoil('observe', () => {
+    const atomA = atom({
+      key: 'observe test a',
+      default: 100,
+    });
+    const atomB = atom({
+      key: 'observe test b',
+      default: 99,
+      effects: [
+        ({observe, setSelf}) => {
+          return observe(atomA, value => {
+            setSelf(value + 1);
+          });
+        },
+      ],
+    });
+
+    const [AtomA, setA, resetA] = componentThatReadsAndWritesAtom(atomA);
+    const [AtomB, setB] = componentThatReadsAndWritesAtom(atomB);
+    const c = renderElements(
+      <>
+        <AtomA />
+        <AtomB />
+      </>,
+    );
+
+    expect(c.textContent).toEqual('10099');
+    act(() => setA(100));
+    expect(c.textContent).toEqual('100101');
+    act(() => setA(101));
+    expect(c.textContent).toEqual('101102');
+  });
+
```

Comment:
Add test cases for resetting and subscribing to selectors.  test when selectors are in error or pending states.

Test this and `onSet()` can only be called directly from the effect function itself, not nested in another callback.

There's an "unsubscribe" unit test that should also test the handler is always unsubscribed same as `onSet()`.

Reply:
@drarmstr  - I ran into an issue with `store.subscribeToTransactions` when passing in the key of a selector. In that case it does not emit events and the observing state doesn't update. I can work around this for selectors by subscribing to the global event stream, but I suspect that will be chattier than what we want, and it breaks a test case regarding resetting an atom that uses onSetValue